src/gcirc.jl

```diff
@@ -1,39 +1,20 @@
-# This file is a part of AstroLib.jl. License is MIT "Expat".
-# Copyright (C) 2016 Mosè Giordano.
-
-function gcirc(units::Integer, ra1::T, dec1::T, ra2::T, dec2::T) where {T<:AbstractFloat}
-    # Convert all quantities to radians.
-    if units == 0
-        # All radians
-        λ_1 = ra1
-        λ_2 = ra2
-        φ_1 = dec1
-        φ_2 = dec2
-    elseif units == 1
-        # Right ascensions are in hours, declinations in degrees.
-        λ_1 = ra1*pi/12.0
-        λ_2 = ra2*pi/12.0
-        φ_1 = deg2rad(dec1)
-        φ_2 = deg2rad(dec2)
-    elseif units == 2
-        # Right ascensions and declinations are in degrees.
-        λ_1 = deg2rad(ra1)
-        λ_2 = deg2rad(ra2)
-        φ_1 = deg2rad(dec1)
-        φ_2 = deg2rad(dec2)
-    else
-        # In any other case throw an error.
-        throw(DomainError(
-            units,
-            "units must be 0 (radians), 1 (hours, degrees) or 2 (degrees)"))
-    end
-    Δφ_2 = (φ_2 - φ_1) * 0.5
-    Δλ_2 = (λ_2 - λ_1) * 0.5
-    Δσ = 2asin(sqrt(abs2(sin(Δφ_2)) + cos(φ_1) * cos(φ_2) * abs2(sin(Δλ_2))))
-    if units == 0
-        return Δσ
-    else
-        return rad2sec(Δσ)
+using SkyCoords
+"""
+AstroCoordinate represents a point in the sky with flexible input units
+"""
+struct AstroCoordinate{T<:AbstractFloat}
+    coord::SkyCoords.ICRSCoords
+
+    function AstroCoordinate(ra::T, dec::T, units::Symbol=:radians) where T<:AbstractFloat
```

Comment:
Why is this an [inner constructor](https://docs.julialang.org/en/v1/manual/constructors/#man-inner-constructor-methods)?